src/main/java/com/es/phoneshop/web/ProductListPageServlet.java

```diff
@@ -22,10 +28,37 @@ public void init(ServletConfig config) throws ServletException {
 
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-        request.setAttribute("products", productService.findProducts());
+        String description = request.getParameter("query");
+        String field = request.getParameter("field");
+        String order = request.getParameter("order");
+        String productToShowPriceHistoryId = request.getParameter("productToShowPriceHistoryId");
+        if (productToShowPriceHistoryId != null) {
+            setProductWithPriceHistoryToRequestById(productToShowPriceHistoryId, request);
+        }
+        request.setAttribute("products", receiveProducts(description, field, order));
         request.getRequestDispatcher("/WEB-INF/pages/productList.jsp").forward(request, response);
     }
 
+    private void setProductWithPriceHistoryToRequestById(String productToShowPriceHistoryId, HttpServletRequest request) {
+        try {
+            long id = Long.parseLong(productToShowPriceHistoryId);
+            Product product = productService.getProduct(id);
+            request.setAttribute("productToShowPriceHistory", product);
+        } catch (NumberFormatException | ProductNotFoundException exception) {
```

Comment:
no need to catch these exceptions
you caught them and did nothing besides printing the stacktrace
it's better and more user-friendly to show some error page when some runtime exception was thrown